### PR TITLE
Add README instructions for OpenBSD installation

### LINKED ISSUE #1148: Abort trap (core dumped) on OpenBSD when connecting via HTTPS
I managed to get Synapse mostly running on my OpenBSD 6.0 server. It seems to work fine for most things, but I can't connect to it over HTTPS. As soon as I try, the console spits out `Abort trap (core dumped)` and the daemon exits. Any ideas?

Here's the log, hostname redacted:

```
2016-09-27 12:05:52,332 - synapse.python_dependencies - 114 - INFO - - Using 'bcrypt' version None from '/home/william/.synapse/lib/python2.7/site-packages/bcrypt/__init__.pyc' to satisfy 'py-bcrypt'
2016-09-27 12:05:52,333 - synapse.python_dependencies - 114 - INFO - - Using 'msgpack' version None from '/home/william/.synapse/lib/python2.7/site-packages/msgpack/__init__.pyc' to satisfy 'msgpack-python>=0.3.0'
2016-09-27 12:05:52,333 - synapse.python_dependencies - 114 - INFO - - Using 'frozendict' version None from '/home/william/.synapse/lib/python2.7/site-packages/frozendict/__init__.pyc' to satisfy 'frozendict>=0.4'
2016-09-27 12:05:52,333 - synapse.python_dependencies - 114 - INFO - - Using 'OpenSSL' version '16.1.0' from '/home/william/.synapse/lib/python2.7/site-packages/OpenSSL/__init__.pyc' to satisfy 'pyopenssl>=0.14'
2016-09-27 12:05:52,333 - synapse.python_dependencies - 114 - INFO - - Using 'pydenticon' version None from '/home/william/.synapse/lib/python2.7/site-packages/pydenticon/__init__.pyc' to satisfy 'pydenticon'
2016-09-27 12:05:52,334 - synapse.python_dependencies - 114 - INFO - - Using 'pymacaroons' version '0.9.0' from '/home/william/.synapse/lib/python2.7/site-packages/pymacaroons/__init__.pyc' to satisfy 'pymacaroons-pynacl'
2016-09-27 12:05:52,334 - synapse.python_dependencies - 114 - INFO - - Using 'service_identity' version '16.0.0' from '/home/william/.synapse/lib/python2.7/site-packages/service_identity/__init__.pyc' to satisfy 'service_identity>=1.0.0'
2016-09-27 12:05:52,334 - synapse.python_dependencies - 114 - INFO - - Using 'signedjson' version '1.0.0' from '/home/william/.synapse/lib/python2.7/site-packages/signedjson/__init__.pyc' to satisfy 'signedjson>=1.0.0'
2016-09-27 12:05:52,334 - synapse.python_dependencies - 114 - INFO - - Using 'blist' version '1.3.6' from '/home/william/.synapse/lib/python2.7/site-packages/blist/__init__.pyc' to satisfy 'blist'
2016-09-27 12:05:52,334 - synapse.python_dependencies - 114 - INFO - - Using 'yaml' version '3.12' from '/home/william/.synapse/lib/python2.7/site-packages/yaml/__init__.pyc' to satisfy 'pyyaml'
2016-09-27 12:05:52,335 - synapse.python_dependencies - 114 - INFO - - Using 'unpaddedbase64' version '1.1.0' from '/home/william/.synapse/lib/python2.7/site-packages/unpaddedbase64.pyc' to satisfy 'unpaddedbase64>=1.1.0'
2016-09-27 12:05:52,335 - synapse.python_dependencies - 114 - INFO - - Using 'ujson' version '1.35' from '/home/william/.synapse/lib/python2.7/site-packages/ujson.so' to satisfy 'ujson'
2016-09-27 12:05:52,335 - synapse.python_dependencies - 114 - INFO - - Using 'nacl' version '0.3.0' from '/home/william/.synapse/lib/python2.7/site-packages/nacl/__init__.pyc' to satisfy 'pynacl==0.3.0'
2016-09-27 12:05:52,335 - synapse.python_dependencies - 114 - INFO - - Using 'nacl.bindings' version '0.3.0' from '/home/william/.synapse/lib/python2.7/site-packages/nacl/__init__.pyc' to satisfy 'pynacl==0.3.0'
2016-09-27 12:05:52,336 - synapse.python_dependencies - 114 - INFO - - Using 'daemonize' version '2.4.7' from '/home/william/.synapse/lib/python2.7/site-packages/daemonize.pyc' to satisfy 'daemonize'
2016-09-27 12:05:52,336 - synapse.python_dependencies - 114 - INFO - - Using 'canonicaljson' version '1.0.0' from '/home/william/.synapse/lib/python2.7/site-packages/canonicaljson.pyc' to satisfy 'canonicaljson>=1.0.0'
2016-09-27 12:05:52,336 - synapse.python_dependencies - 114 - INFO - - Using 'twisted' version '16.4.1' from '/home/william/.synapse/lib/python2.7/site-packages/twisted/__init__.pyc' to satisfy 'Twisted>=15.1.0'
2016-09-27 12:05:52,336 - synapse.python_dependencies - 114 - INFO - - Using 'saml2' version '3.0.2' from '/home/william/.synapse/lib/python2.7/site-packages/saml2/__init__.pyc' to satisfy 'pysaml2>=3.0.0,<4.0.0'
2016-09-27 12:05:52,337 - synapse.python_dependencies - 114 - INFO - - Using 'pyasn1' version '0.1.9' from '/home/william/.synapse/lib/python2.7/site-packages/pyasn1/__init__.pyc' to satisfy 'pyasn1'
2016-09-27 12:05:52,337 - synapse.python_dependencies - 114 - INFO - - Using 'PIL' version None from '/home/william/.synapse/lib/python2.7/site-packages/PIL/__init__.pyc' to satisfy 'pillow'
2016-09-27 12:05:52,383 - synapse.app.homeserver - 290 - INFO - - Server hostname: hostname.example.com
2016-09-27 12:05:52,385 - synapse.app.homeserver - 291 - INFO - - Server version: Synapse/0.18.0
2016-09-27 12:05:52,394 - synapse.app.homeserver - 309 - INFO - - Preparing database: sqlite3...
2016-09-27 12:05:52,401 - synapse.storage.prepare_database - 212 - INFO - - Upgrading schema to v35
2016-09-27 12:05:52,403 - synapse.app.homeserver - 327 - INFO - - Database prepared in sqlite3.
2016-09-27 12:05:52,403 - synapse.server - 144 - INFO - - Setting up.
2016-09-27 12:05:52,411 - synapse.storage.event_push_actions - 465 - INFO - - Searching for stream ordering 1 month ago
2016-09-27 12:05:52,411 - synapse.storage.event_push_actions - 471 - INFO - - Found stream ordering 1 month ago: it's 0
2016-09-27 12:05:52,419 - synapse.server - 146 - INFO - - Finished setting up.
2016-09-27 12:05:52,464 - synapse.util.httpresourcetree - 43 - INFO - - Attaching <twisted.web.resource.EncodingResourceWrapper object at 0x7544efcc> to path /_matrix/client/r0
2016-09-27 12:05:52,464 - synapse.util.httpresourcetree - 43 - INFO - - Attaching FilePath('/home/william/.synapse/lib/python2.7/site-packages/synapse/static') to path /_matrix/static
2016-09-27 12:05:52,464 - synapse.util.httpresourcetree - 43 - INFO - - Attaching <twisted.web.resource.EncodingResourceWrapper object at 0x7544efcc> to path /_matrix/client/versions
2016-09-27 12:05:52,464 - synapse.util.httpresourcetree - 43 - INFO - - Attaching <twisted.web.resource.EncodingResourceWrapper object at 0x7544efcc> to path /_matrix/client/unstable
2016-09-27 12:05:52,465 - synapse.util.httpresourcetree - 43 - INFO - - Attaching <synapse.rest.media.v1.media_repository.MediaRepositoryResource instance at 0x7545722c> to path /_matrix/media/v1
2016-09-27 12:05:52,465 - synapse.util.httpresourcetree - 43 - INFO - - Attaching <twisted.web.resource.EncodingResourceWrapper object at 0x7544efcc> to path /_matrix/client/api/v1
2016-09-27 12:05:52,465 - synapse.util.httpresourcetree - 43 - INFO - - Attaching <synapse.rest.media.v0.content_repository.ContentRepoResource instance at 0x7546034c> to path /_matrix/content
2016-09-27 12:05:52,465 - synapse.util.httpresourcetree - 43 - INFO - - Attaching <synapse.rest.key.v2.KeyApiV2Resource instance at 0x7543c6ac> to path /_matrix/key/v2
2016-09-27 12:05:52,466 - synapse.util.httpresourcetree - 43 - INFO - - Attaching <twisted.web.resource.EncodingResourceWrapper object at 0x7544efcc> to path /_matrix/client/v2_alpha
2016-09-27 12:05:52,466 - synapse.util.httpresourcetree - 43 - INFO - - Attaching FilePath('/home/william/.synapse/lib/python2.7/site-packages/syweb/webclient') to path /_matrix/client
2016-09-27 12:05:52,466 - synapse.util.httpresourcetree - 43 - INFO - - Attaching <synapse.rest.key.v1.server_key_resource.LocalKey instance at 0x7545704c> to path /_matrix/key/v1
2016-09-27 12:05:52,466 - synapse.util.httpresourcetree - 43 - INFO - - Attaching <synapse.rest.media.v1.media_repository.MediaRepositoryResource instance at 0x7545722c> to path /_matrix/media/r0
2016-09-27 12:05:52,467 - synapse.util.httpresourcetree - 43 - INFO - - Attaching <synapse.federation.transport.server.TransportLayerServer object at 0x754571ec> to path /_matrix/federation/v1
2016-09-27 12:05:52,469 - twisted - 154 - INFO - - SynapseSite (TLS) starting on 8448
2016-09-27 12:05:52,470 - twisted - 154 - INFO - - Starting factory <synapse.http.site.SynapseSite instance at 0x7543cd8c>
2016-09-27 12:05:52,470 - synapse.app.homeserver - 199 - INFO - - Synapse now listening on port 8448
2016-09-27 12:05:52,471 - synapse.app.homeserver - 31 - INFO - - Set file limit to: 1024
2016-09-27 12:05:52,473 - synapse.storage.background_updates - 94 - INFO - - Starting background schema updates
2016-09-27 12:05:52,504 - synapse.metrics - 212 - INFO - - Collecting gc 0
2016-09-27 12:05:52,540 - synapse.push.pusherpool - 186 - INFO - - Starting 0 pushers
2016-09-27 12:05:52,540 - synapse.push.pusherpool - 205 - INFO - - Started pushers
2016-09-27 12:05:53,484 - synapse.storage.background_updates - 115 - INFO - - No more background updates to do. Unscheduling background update task.
Abort trap (core dumped)
```

### PR DESCRIPTION
**Pull request by williamleuschner**
Signed-off-by: William Leuschner <wel2138@rit.edu>

Closes #1148

### CODE REVIEW

**Comment by williamleuschner** [comment]
Wow, @matrixbot; I think they got it...

File: README.rst [hunk 1]
```diff
@@ -364,6 +368,36 @@ Synapse can be installed via FreeBSD Ports or Packages contributed by Brendan Mo
...
+OpenBSD
+-------
+
+There is currently no port for OpenBSD. Additionally, OpenBSD's security
+settings require a slightly more difficult installation process.
+
+1) Ensure that you have the Bourne Again Shell (``bash``) installed. The Korn
+   Shell (``ksh``), which is OpenBSD's default shell, does not support the
+   ``source`` command necessary to activate the python virtualenv to be set up
+   later.
```
Comment (qbit) [inline README.rst]: ksh does support "sourcing" files - it just uses the `.` syntax: `. bin/activate` for example.

**Comment by qbit** [comment]
Also here is an example rc.d script:

```
cat /etc/rc.d/synapse                                                                                                                                                                        
#!/bin/sh

datadir="/usr/local/_synapse/.synapse"
daemon_flags="/usr/local/_synapse/.synapse/homeserver.yaml"
daemon="${datadir}/bin/synctl"
daemon_user="_synapse"

. /etc/rc.d/rc.subr
. ${datadir}/bin/activate

rc_reload=NO
rc_usercheck=NO

pexp="/usr/local/_synapse/.synapse/bin/python.*"

rc_start() {
        ${rcexec} "${daemon} start ${daemon_flags}"
}

rc_stop() {
        ${rcexec} "${daemon} stop ${daemon_flags}"
}

rc_restart() {
        ${rcexec} "${daemon} restart ${daemon_flags}"
}

rc_cmd $1
```

**Comment by erikjohnston** [comment]
Thanks for this! Does @qbit's suggestion of `. bin/activate` work?

File: README.rst (hunk 1 above)
Comment (williamleuschner) [inline README.rst]: Ah, I didn't know that, and I guess I just forgot to try it when I was setting it up. Thanks!

**Comment by williamleuschner** [comment]
@erikjohnston Yes, it does. I've tested it and updated my README changes accordingly.

File: README.rst [hunk 2]
```diff
@@ -364,6 +368,32 @@ Synapse can be installed via FreeBSD Ports or Packages contributed by Brendan Mo
...
  - Packages: ``pkg install py27-matrix-synapse``
 
+
+OpenBSD
+-------
+
+There is currently no port for OpenBSD. Additionally, OpenBSD's security
+settings require a slightly more difficult installation process.
+
+2) Create a new directory in ``/usr/local`` called ``_synapse``. Also, create a
```
Comment (erikjohnston) [inline README.rst]: Might want to start at 1 :)

Comment (williamleuschner) [inline README.rst]: I completely forgot that.. Whoops!

**Pull request merged**